solutionbox/code_free_ml/mltoolbox/code_free_ml/data/trainer/feature_transforms.py

```diff
@@ -0,0 +1,713 @@
+        with file_io.FileIO(checkpoint, 'r') as f_in, file_io.FileIO(checkpoint_tmp, 'w') as f_out:
+          f_out.write(f_in.read())
+        with tf.Session() as sess:
+          saver.restore(sess, checkpoint_tmp)
+          output_graph_def = tf.graph_util.convert_variables_to_constants(sess,
+                                                                          g.as_graph_def(),
+                                                                          [so.op.name])
+        file_io.delete_file(checkpoint_tmp)
+      tensors_out = tf.import_graph_def(output_graph_def,
+                                        input_map={si.name: tensor_in},
+                                        return_elements=[so.name])
+      return tensors_out[0]
+
+    return _tito_out
+
+  return _tito_from_checkpoint(_image_to_vec,
+                               INCEPTION_V3_CHECKPOINT,
+                               INCEPTION_EXCLUDED_VARIABLES)
+# ------------------------------------------------------------------------------
+# ------------------------------------------------------------------------------
+# end of transform functions
+# ------------------------------------------------------------------------------
+# ------------------------------------------------------------------------------
+
+
+def make_preprocessing_fn(output_dir, features, keep_target):
+  """Makes a preprocessing function.
+
+  Args:
+    output_dir: folder path that contains the vocab and stats files.
+    features: the features dict
+
+  Returns:
+    a function that takes a dict of input tensors
+  """
+  def preprocessing_fn(inputs):
+    """Preprocessing function.
+
+    Args:
+      inputs: dictionary of raw input tensors
+
+    Returns:
+      A dictionary of transformed tensors
+    """
+    stats = json.loads(
+      file_io.read_file_to_string(
+          os.path.join(output_dir, STATS_FILE)).decode())
+
+    result = {}
+    for name, transform in six.iteritems(features):
+      transform_name = transform['transform']
+
+      if transform_name == KEY_TRANSFORM:
+        transform_name = 'identity'
+      elif transform_name == TARGET_TRANSFORM:
+        if not keep_target:
+          continue
+        if file_io.file_exists(os.path.join(output_dir, VOCAB_ANALYSIS_FILE % name)):
+          transform_name = 'one_hot'
+        else:
+          transform_name = 'identity'
+
+      if transform_name == 'identity':
+        result[name] = inputs[name]
+      elif transform_name == 'scale':
+        result[name] = scale(
+            inputs[name],
+            min_x_value=stats['column_stats'][name]['min'],
+            max_x_value=stats['column_stats'][name]['max'],
+            output_min=transform.get('value', 1) * (-1),
+            output_max=transform.get('value', 1))
+      elif transform_name in [ONE_HOT_TRANSFORM, EMBEDDING_TRANSFROM,
+                              TFIDF_TRANSFORM, BOW_TRANSFORM]:
+        vocab_str = file_io.read_file_to_string(
```

Comment:
We need the file in memory as we pass it to string_to_index_table_from_tensor. If we used index_table_from_file, we would not need to read it directly, but we would have to change the format of your vocab files (one line in our vocab file is "token,example_count"). In any event, the vocab has to be in memory at some point to do the lookup, right?

What is your suggestion with file_io.FileIO? Do you want me to loop over the file line-by-line, and split the columns into two lists manually?